src/Core/CFamily/IRequest.cs

```diff
@@ -35,6 +36,11 @@ public interface IRequest
         /// </summary>
         void WriteRequest(BinaryWriter writer);
 
+        /// <summary>
+        /// Any environment variables that need to be passed to the subprocess. Can be null.
+        /// </summary>
+        IDictionary<string, string> EnvironmentVariables { get; }
```

Comment:
The new protocol requires some inputs to be passed via environment variables, so I've added a dictionary to the request interface.